src/init/AppRunner.ts

```diff
@@ -56,97 +50,77 @@ export class AppRunner {
       dumpErrorState: true,
       logLevel: params.loggingLevel as LogLevel,
     };
-    const configFile = resolveAssetPath(params.config);
 
-    // Create and execute the app
-    this.createApp(loaderProperties, configFile, params)
-      .then(
-        async(app): Promise<void> => app.start(),
-        (error: Error): void => {
-          // Instantiation of components has failed, so there is no logger to use
-          stderr.write(`Error: could not instantiate server from ${configFile}\n`);
-          stderr.write(`${error.stack}\n`);
-          process.exit(1);
-        },
-      ).catch((error): void => {
-        this.logger.error(`Could not start server: ${error}`, { error });
-        process.exit(1);
-      });
+    // Create varResolver
+    const varConfigFile = resolveAssetPath(params.varConfig);
+    this.createComponent(
+      loaderProperties as IComponentsManagerBuilderOptions<VarResolver>, varConfigFile, {}, varConfigComponentIri,
+    ).then(
+    //  Using varResolver resolve vars and start app
+      async(varResolver): Promise<void> => {
+        let vars: VarRecord;
+        try {
+          vars = await varResolver.handle(argv);
+        } catch (error: unknown) {
+          this.exitWithError(error as Error, 'Error in computing variables', stderr);
+        }
+        return this.initApp(
+          loaderProperties, resolveAssetPath(params.config), vars as unknown as Record<string, any>, stderr,
+        );
+      },
+
+      (error): void => this.exitWithError(
+        error, `Error in loading variable configuration from ${varConfigFile}`, stderr,
+      ),
+    ).catch((error): void => this.exitWithError(error, 'Could not start the server', stderr));
   }
 
-  /**
-   * Creates the main app object to start the server from a given config.
-   * @param loaderProperties - Components.js loader properties.
-   * @param configFile - Path to a Components.js config file.
-   * @param variables - Variables to pass into the config file.
-   */
-  public async createApp(
+  private exitWithError(error: Error, message: string, stderr: WriteStream): never {
+    stderr.write(`message\n`);
+    stderr.write(`${error.stack}\n`);
+    process.exit(1);
+  }
+
+  private async initApp(
     loaderProperties: IComponentsManagerBuilderOptions<App>,
-    configFile: string,
-    variables: CliParams | Record<string, any>,
-  ): Promise<App> {
-    // Translate command-line parameters if needed
-    if (typeof variables.loggingLevel === 'string') {
-      variables = this.createVariables(variables as CliParams);
+    configFile: string, vars: Record<string, any>, stderr: WriteStream,
+  ): Promise<void> {
+    let app: App;
+    // Create app
+    try {
+      app = await this.createComponent(loaderProperties, configFile, vars, appComponentIri);
+    } catch (error: unknown) {
+      this.exitWithError(error as Error, `Error: could not instantiate server from ${configFile}`, stderr);
```

Comment:
just throw this

Reply:
Here too, message provides clarity on whether process exited due to failure in componentsjs-instantiation of app or initialization of app later. So that developer can check their configuration/implementation accordingly. Blanket error with out that context will not capture that distinction.

Reply:
Agreed, so:

```suggestion
  throw new Error(`Could not instantiate server from ${configFile}: ${error.message}`);
```

Reply:
That way we will loose original error stack pin pointing the cause. We can create simple custom error, and preserve cause stack.
Better will be to use popular [verror](https://www.npmjs.com/package/verror) minimal dependency, if allowed to add one.